skyreader/result.py

```diff
@@ -537,7 +537,13 @@ def serialize(self) -> PyDictResult:
             for key in nside_data.dtype.metadata:
                 # dtype.metadata is a mappingproxy (dict-like) containing numpy-typed values
                 # convert numpy types to python bultins to be JSON-friendly
-                pydict[nside]['metadata'][key] = nside_data.dtype.metadata[key].item()
+                val = nside_data.dtype.metadata[key]
+                if type(val).__module__ == np.__name__ :
```

Comment:
Is there a way to do this with `isinstance()`?

Reply:
I would need to assume the data type(s) beforehand, or loop over all the possible data types to find a match.

An alternative would be to use `try / except AttributeError` but I guess exceptions should not be used as surrogates for conditional statements.

Reply:
Is there a common numpy base class for all its special data types?

Reply:
can you use https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.number?

Reply:
This would only catch numeric types. I believe `numpy` may wrap strings and dates as well?

Reply:
Alright, with `isinstance(val, np.generic)` it works!